opensir/models/model.py

```diff
@@ -194,3 +206,10 @@ def function_handle(t, *par_fit, pop=population):
     def _update_ic(self):
         """ updates initial conditions if necessary """
         return self
+
+def _validate_params(arr, num_params):
+        if np.isnan(arr).any() or len(arr) != num_params:
+            raise Model.InvalidNumberOfParametersError
+
+        if sum(arr < 0) > 0:
```

Comment:
Sorry, I am not familiar enough with Python. What is this line doing exactly?

I know you are adding through the array, but I am curious about the `arr < 0` part.

Reply:
So arr < 0 yields a boolean array with 1 for each element of arr that's non-zero. Summing over that will produce the number of negative parameters. Then, the exception is raised if any of the elements of arr is negative.